src/index.ts

```diff
@@ -139,9 +139,10 @@ function urlcatImpl(
   const { renderedPath, remainingParams } = path(pathTemplate, params);
   const cleanParams = removeNullOrUndef(remainingParams);
   const renderedQuery = query(cleanParams, config);
-  const pathAndQuery = join(renderedPath, '?', renderedQuery);
+  const pathAndQuery = join(renderedPath.trim(), '?', renderedQuery);
+
   return baseUrl
-    ? join(baseUrl, '/', pathAndQuery)
+    ? renderedPath.trim().length ? join(baseUrl, '/', pathAndQuery) : join(baseUrl, '?', pathAndQuery)
```

Comment:
This solution is correct but I find nested ternary operators hard to read. Especially when they aren't parenthesized, like above. Could you please extract this line into a separate function?

Reply:
Also I'm not sure I understand the purpose of the `.trim()` calls but it may be just because I'm tired. Is this just for convenience or do you think there is a valid use case?

Reply:
> Also I'm not sure I understand the purpose of the `.trim()` calls but it may be just because I'm tired. Is this just for convenience or do you think there is a valid use case?
@balazsbotond  here I thought that what if someone pass white space 
 ```urlcat('http://example.com/path', " ", { p: 'a' });``` this will not work as expected.

Reply:
Thanks, so it's for convenience. What we need to decide now is whether we should do it this way.

1. If we don't trim the string, the resulting URL would be invalid
    1. one possible course or action is to throw an error
    2. another one is to just build the incorrect URL and let its consumer (e.g. an AJAX request) thrown an error
2. If we silently trim it, this might lead to unexpected results if some unknowingly passes an invalid path with leading and trailing whitespace as a result of a programming error and we try to make sense of it and trim it.

I'm generally in favor of being explicit and failing early - this might be a bit inconvenient for the caller but leads to safer software. But I also think urlcat shouldn't do more validation than necessary for it to correctly function. So I think (1./ii.) would be a good middle ground. What do you think?

Reply:
@balazsbotond Perfect brother !! I also think this is extra validation.We shouldn't do that and let consumer do there work !! 
I will change it 👍🏼

Reply:
Thanks!

Reply:
Thanks!